packages/dapp/src/hooks/usePopperIsOpened.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+import { useMenuStore } from '../stores';
+
+export const usePopperIsOpened = () => {
+  const [popperOpened, setPopperOpened] = useState<boolean>();
+  const [
+    openMainNavbarMenu,
+    openNavbarWalletMenu,
+    openNavbarSubMenu,
+    openNavbarChainsMenu,
+    openNavbarWalletSelectMenu,
+  ] = useMenuStore((state) => [
+    state.openMainNavbarMenu,
+    state.openNavbarWalletMenu,
+    state.openNavbarSubMenu,
+    state.openNavbarChainsMenu,
+    state.openNavbarWalletSelectMenu,
+  ]);
+
+  useEffect(() => {
+    setPopperOpened(
+      openMainNavbarMenu ||
+        openNavbarWalletMenu ||
+        openNavbarSubMenu !== 'None' ||
+        openNavbarChainsMenu ||
+        openNavbarWalletSelectMenu,
+    );
+  }, [
+    openMainNavbarMenu,
+    openNavbarChainsMenu,
+    openNavbarSubMenu,
+    openNavbarWalletMenu,
+    openNavbarWalletSelectMenu,
+  ]);
+
+  return popperOpened;
+};
```

Comment:
I have used a hook for listening on state changes of our menus. 
@chybisov @Abhikumar98 @0xsebn is this too much or good to go? Wdyt?

Reply:
LGTM

a minor suggestion
```ts
useEffect(() => {
    const isPopperOpened = openMainNavbarMenu ||
        openNavbarWalletMenu ||
        openNavbarSubMenu !== 'None' ||
        openNavbarChainsMenu ||
        openNavbarWalletSelectMenu;

    setPopperOpened(isPopperOpened);
}, [ ... ])
};
```

Reply:
@dennyscode this is too much and can be optimized. I suggest ->

```ts
export const usePopperIsOpened = () => {
  const [
    openMainNavbarMenu,
    openNavbarWalletMenu,
    openNavbarSubMenu,
    openNavbarChainsMenu,
    openNavbarWalletSelectMenu,
  ] = useMenuStore((state) => [
    state.openMainNavbarMenu,
    state.openNavbarWalletMenu,
    state.openNavbarSubMenu,
    state.openNavbarChainsMenu,
    state.openNavbarWalletSelectMenu,
  ]);

  return openMainNavbarMenu ||
        openNavbarWalletMenu ||
        openNavbarSubMenu !== 'None' ||
        openNavbarChainsMenu ||
        openNavbarWalletSelectMenu;
};
```

This hook does exactly the same and has -1 re-render.

Reply:
Great feedback! Thank you guys